app/models/user.rb

```diff
@@ -84,4 +86,8 @@ def can_manage?(comment)
   def watching?(task)
     self.tasks.where(id: task.id).exists?
   end
+
+  def with_avatar?
+    self.avatar.attached? && self.avatar.attachment.blob.present? && self.avatar.attachment.blob.persisted?
```

Comment:
you don't need `self` here

Reply:
removed

Reply:
not removed

Reply:
fyi:
`avatar&.attachment&.blob&.persisted?` should be enough to validate this

Reply:
this is needed Roman, this is an issue in Rails 5.2 (fixed in Rails 6) ... so code is like this (it's came from me)

Reply:
in this case `avatar&.attachment&.blob&.persisted?` should be enough